crates/bevy_render/src/render_phase/mod.rs

```diff
@@ -84,10 +84,13 @@ impl<I: PhaseItem> RenderPhase<I> {
         let mut draw_functions = draw_functions.write();
         draw_functions.prepare(world);
 
-        for item in &self.items {
-            let draw_function = draw_functions.get_mut(item.draw_function()).unwrap();
-            draw_function.draw(world, render_pass, view, item);
-        }
+        self.items
+            .iter()
+            .filter(|item| !item.skip())
```

Comment:
This feels like it would be significantly slower than maintaining an index variable and skipping over large numbers of items at once based on the batch size.